### ISSUE TITLE
Unable to find compiler, causes exception

### ISSUE THREAD

**Issue by mattgodbolt** [mattgodbolt/compiler-explorer#771]
https://sentry.io/compiler-explorer/compiler-explorer/issues/434581210/

```
TypeError: Cannot read property 'lang' of undefined
  at langOfCompiler (compiler.js:1253:1)
  at new i (compiler.js:82:1)
  at compilerFactory (hub.js:158:1)
  at new <anonymous> (hub.js:82:1)
  at new e.items.Component (ext/golden-layout/dist/goldenlayout.js:3706:1)
...
(19 additional frame(s) were not displayed)
```

**Comment by mattgodbolt** [comment]
Looks like problems around:

```javascript
    Compiler.prototype.langOfCompiler = function (compilerId) {
        var compiler = _.find(options.compilers, function (compiler) {
            return compiler.id === compilerId;
        });
        return compiler.lang;  /// <--------- here
    };
```

**Comment by mattgodbolt** [comment]
Doesn't seem to have happened before today's update

**Comment by mattgodbolt** [comment]
Added some workaround code in a212ba77de607e25251be9b60fa795192024507e but not sure why it's happening.

**Comment by RabsRincon** [comment]
Relevant https://sentry.io/compiler-explorer/compiler-explorer/issues/434620501/

**Comment by mattgodbolt** [comment]
From #772:

Looks like an issue where if a URL has a pathname instead of an ID then the client code can't find it in the list of compilers. We'll have to support aliases too.

https://sentry.io/compiler-explorer/compiler-explorer/issues/434963846/

Unable to find compiler id "/opt/clang+llvm-3.4.1-x86_64-unknown-ubuntu12.04/bin/clang++"

**Comment by mattgodbolt** [comment]
Looks like we lost a Go compiler along the way. Maybe we need to add an alias?

https://sentry.io/compiler-explorer/compiler-explorer/issues/435919473/

> Unable to find compiler id "gccgo491"

go compilers:

```
gccgo494      | x86 gccgo 4.9.4
gccgo630      | x86 gccgo 6.3.0
gccgo720      | x86 gccgo 7.2.0
```

**Comment by RabsRincon** [comment]
Fixed by 4e5aef0db1fe58c7383b99b349cbe858f8b819e2 . Sorry..
